clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/server/P2PFileTransferServerHandler.java

```diff
@@ -101,9 +107,24 @@ protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest httpReque
     }
     final BlobTransferPayload blobTransferRequest;
     final File snapshotDir;
+    boolean isHybridStore;
+
     try {
       blobTransferRequest = parseBlobTransferPayload(URI.create(httpRequest.uri()));
       snapshotDir = new File(blobTransferRequest.getSnapshotDir());
+      isHybridStore = blobSnapshotManager.isStoreHybrid(blobTransferRequest.getStoreName());
+
+      // recreate snapshot for hybrid store
+      if (isHybridStore) {
+        try {
+          BlobTransferPartitionMetadata metadataForHybrid = prepareMetadata(blobTransferRequest, ctx);
```

Comment:
nitpick on the code style.
We don't need to pass `ctx` to every downstream methods. We can just throw exceptions and catch it in this main method. Once caught, we can `setupResponseAndFlush` accordingly. It can make the codes more clean. As we are adding more if-else conditions, this method will become clumsy and a bit hard to read